### ISSUE TITLE
   The system cannot move files to a different disk drive.。

### ISSUE THREAD

**Issue by laterdayi** [bsutton/simpleyaml#3]
Cannot rename file to 'E:\projects\mobile\pubspec.yaml', path = 'C:\Users\admin\AppData\Local\Temp\e9d1e8f8-fb7c-48ac-ab3d-cd6a66781362.tmp' (OS Error: The system cannot move files to a different disk drive.

**Comment by bsutton** [comment]
You are going to have to provide more information than that.

**Comment by laterdayi** [comment]
```
  final settingsYaml = SettingsYaml.load(pathToSettings: '$projectPath\\pubspec.yaml');
```
```
  String appName = settingsYaml['appName'];
    stdout.write('重命名 $appName yaml 版本号\n');
    String oldVersion = settingsYaml['version'];
    String newVersion ='2.2.0';
    settingsYaml['version'] = newVersion;
    print(settingsYaml.valueMap);
    await settingsYaml.save();
```


My project is on disk E

**Comment by bsutton** [comment]
So save is doing a rename which doesn't work across drives.

The exception needs to be caught in the save method and then a copy/deleted
performed.

The fastest way for a fix at the moment would be if you raised a pr.

**Comment by laterdayi** [comment]
Do I need to actively delete and save manually? I am not familiar with dart, may I ask when it can be repaired

**Comment by bsutton** [comment]
No the save method will need to be changed.

I'm unlikely to have time in the next few weeks, hence the suggestion of
raising a PR.

If you want to have a go fixing it, it should be a fairly easy fix.

**Comment by laterdayi** [comment]
Just add a try catch to save ？ If it's easy, I'll try PR

**Comment by bsutton** [comment]
Yes and in the catch block perform a copy and delete if the file that was
being renamed/moved.

**Comment by bsutton** [comment]
oops, this was fixed in 8.1.0 and I failed to update this issue and I've been working on the wrong branch.

So apologies for wasting your time.

I've now merged my work into master. If you upgraded to 8.1.0  you should have the fix.

sorry!!!!!
